src/pages/ListScholarships.jsx

```diff
@@ -57,6 +45,30 @@ function ListScholarships() {
     hideExpired: true,
   };
 
+  const filterChips = {};
+  if (Number.isInteger(minAmount)) {
+    filterChips[`Min $${minAmount}`] = () =>
+      setQueryParams({ minAmount: undefined });
+  }
+  if (Number.isInteger(maxAmount)) {
+    filterChips[`Max $${maxAmount}`] = () =>
+      setQueryParams({ maxAmount: undefined });
+  }
+  majors?.forEach(
+    (m) =>
+      (filterChips[m] = () =>
+        setQueryParams({
+          majors: majors?.filter((major) => major !== m),
+        }))
+  );
+  grades?.forEach(
+    (g) =>
+      (filterChips[GradeLevel.toString(g)] = () =>
+        setQueryParams({
+          grades: grades?.filter((grade) => grade !== g),
+        }))
```

Comment:
How ugly would it look if we mapped the chip name to the query param arguments to apply?
```suggestion
      (filterChips[GradeLevel.toString(g)] =({
          grades: grades?.filter((grade) => grade !== g),
        }))
```

Below in the Chip, we could do `onClick={() => setQueryParams(queryParams)}`.

Though if we do this, we should probably use better names since `queryParams` and `setQueryParams` sound like they set/use the whole thing. Maybe `updateQueryParams`? Or we could keep the `setQueryParams` name, have it replace everything by default (so "set" = "create or replace" by default) and accept an `options` object parameter that's explicit about the merging behavior. E.g. `setQueryParams(queryParams, {merge: true})`

Reply:
^ I tried to implement this but it would cause it to crash. It complained about being an unstable operation.